src/canopy/chat_engine/chat_engine.py

```diff
@@ -197,8 +197,10 @@ def chat(self,
             messages,
             max_tokens=self.max_prompt_tokens
         )
+        max_tokens = model_params.get("max_tokens", self.max_generated_tokens) if model_params else self.max_generated_tokens
+        model_params = {k: v for k, v in model_params.items() if k != 'max_tokens'} if model_params else None
         llm_response = self.llm.chat_completion(llm_messages,
```

Comment:
Should we be updating `BaseLLM` to move the `max_tokens` and `stream` param inside of `model_params`?

Or should we go the other way and pull more params out of `model_params`? Or neither?